File: Backend/src/main/java/fr/rowlaxx/quizzerai/question/Question.java
package fr.rowlaxx.quizzerai.question;

import lombok.Data;

import java.util.List;
import java.util.Map;
import java.util.UUID;

@Data
public class Question {

    private String question;
    private List<String> options;
    private double multiplicator;
    private int answer;

    private Map<UUID, Answer> answers;
}

File: Frontend/quiz_app/lib/dto/question.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:quiz_app/dto/answer.dart';

part 'question.freezed.dart';
part 'question.g.dart';


@freezed
class Question with _$Question {

  const factory Question({
    required String question,
    required List<String> options,
    required double multiplicator,
    required int answer,
    required Map<String, Answer> answers
  }) = _Question;

  factory Question.fromJson(Map<String, Object?> json) => _$QuestionFromJson(json);

}